src/test/java/reposense/model/RepoLocationTest.java

```diff
@@ -0,0 +1,115 @@
+        assertValidLocation(repoUrl, repoUrl, "RepoSense", "reposense", null);
+
+        String repoUrlWithBranch = "https://github.com/reposense/testrepo-Alpha.git#release";
+        assertValidLocation(repoUrlWithBranch, "https://github.com/reposense/testrepo-Alpha.git",
+                "testrepo-Alpha", "reposense", "release");
+    }
+
+    @Test
+    public void repoLocation_parseInvalidRepoUrl_throwsInvalidLocationException() {
+        // non GitHub url should rejected
+        assertInvalidLocation("https://gitlab.com/reposense/RepoSense.git");
+        // url without organisation name should be rejected
+        assertInvalidLocation("https://github.com/reposense.git");
+        // url without repo name should be rejected
+        assertInvalidLocation("https://github.com/reposense/.git");
+        assertInvalidLocation("https://github.com/reposense");
+        // url without a .git suffix should be rejected
+        assertInvalidLocation("https://github.com/reposense/RepoSensegit");
+        assertInvalidLocation("https://github.com/reposense/RepoSense-git");
+        assertInvalidLocation("https://github.com/reposense/RepoSense.gi");
+        assertInvalidLocation("https://github.com/reposense/RepoSense");
+    }
+
+    @Test
+    public void repoLocation_parseBranchUrl_success() throws Exception {
+        String branchUrl = "https://github.com/reposense/RepoSense/tree/feature_branch_issue#1010";
+        assertValidLocation(branchUrl, "https://github.com/reposense/RepoSense.git",
+                 "RepoSense", "reposense", "feature_branch_issue#1010");
+    }
+
+    @Test
+    public void repoLocation_parseInvalidBranchUrl_throwsInvalidLocationException() {
+        // ftp url should be rejected
+        assertInvalidLocation("ftp://github.com/reposense/RepoSense/tree/feature_branch_issue#1010");
+        // url without branch name should be rejected
+        assertInvalidLocation("https://github.com/reposense/RepoSense/tree/");
+        assertInvalidLocation("https://github.com/reposense/RepoSense/tree");
+        // url without the 'tree' keyword should be rejected
+        assertInvalidLocation("https://github.com/reposense/RepoSense/feature_branch_issue#1010");
+        assertInvalidLocation("https://github.com/reposense/RepoSense/TREE/feature_branch_issue#1010");
+        assertInvalidLocation("https://github.com/reposense/RepoSense/tre/feature_branch_issue#1010");
+        // non GitHub url should be rejected
+        assertInvalidLocation("https://gitlab.com/reposense/RepoSense/tree/feature_branch_issue#1010");
+    }
+
+    @Test
+    public void repoLocation_parsePath_success() throws Exception {
+        Method tryParsingAsPath = RepoLocation.class.getDeclaredMethod("tryParsingAsPath", String.class,
+                boolean.class);
+        tryParsingAsPath.setAccessible(true);
+
+        String pathToGitDirectory = Paths.get("home", "users", "tom", "Desktop",
```

Comment:
For this test, I am using `Paths.get` to create the path used for testing.
This ensures that the path created is valid on whichever OS is used to run the test.
This way, the test can be run on any OS.